jina/helper.py

```diff
@@ -909,6 +909,48 @@ def __delete__(self, obj):
             del obj.__dict__[f'CACHED_{self.func.__name__}']
 
 
+class _cache_invalidate(object):
+    """Class for cache invalidation, remove strategy.
+
+    :param func: func to wrap as a decorator.
+    :param func_to_invalidate: String as the function name to invalidate cached
+        data. E.g. in :class:`cached_property` we cache data inside the class obj
+        with the `key`: `CACHED_{func.__name__}`, the func name in `cached_property`
+        is the name to invalidate.
+    """
+
+    def __init__(self, func, func_to_invalidate: str):
```

Comment:
func_to_invalidate or `property` or `attribute` right? I think `attribute` is more generic

Reply:
@JoanFM you mean renamed the variable?

Reply:
yes